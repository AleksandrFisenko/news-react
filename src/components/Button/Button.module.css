.button {
  width: 100%;
  height: 30px;
  color: var(--header-title-color);
  background-color: var(--add-button-color);
  font-weight: 600;
  font-size: 1rem;
  border-radius: 10px;
  border-style: none;
}

.button:hover {
  background-color: var(--add-button-color-hover);
}
