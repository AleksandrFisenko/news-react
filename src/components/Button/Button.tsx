import css from "./Button.module.css";

interface ButtonProps {
  children: React.ReactNode;
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  disabled?: boolean;
  className?: string;
  type?: "button" | "submit" | "reset";
}

const Button = ({
  type = "button",
  children,
  onClick,
  disabled = false,
  className,
}: ButtonProps) => {
  return (
    <button
      type={type}
      onClick={onClick}
      disabled={disabled}
      className={`${css.button} ${className}`}
    >
      {children}
    </button>
  );
};

export default Button;
