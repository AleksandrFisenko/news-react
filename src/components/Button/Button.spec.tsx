import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import Button from "./Button";
import userEvent from "@testing-library/user-event";

describe('Button', () => {
  it('should be render buttin', () => {
    render(
      <Button>АВЕ НЛМК</Button>
    );

    const button = screen.queryByRole('button', { name: 'АВЕ НЛМК' });
    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
  });

  it('should be call onClick', async () => {
    const handleClick = vi.fn();
    render(
      <Button onClick={handleClick}>АВЕ НЛМК</Button>
    );

    const button = screen.getByRole('button', { name: 'АВЕ НЛМК' });
    
    await userEvent.click(button);

    expect(handleClick).toHaveBeenCalledOnce();
  });

  it('should be render buttin', () => {
    render(
      <Button>АВЕ НЛМК</Button>
    );

    const button = screen.queryByRole('button', { name: 'АВЕ НЛМК' });
    expect(button).toBeInTheDocument();
  });

  it ('should be render a disabled button', () => {
    render(
      <Button disabled>АВЕ НЛМК</Button>
    );

    const button = screen.queryByRole('button', { name: 'АВЕ НЛМК' });
    expect(button).toBeInTheDocument();
    expect(button).toBeDisabled();
    expect(button).toHaveAttribute("disabled");
  })

  it ('should not call onClick when disabled', async () => {
    const handleClick = vi.fn();
    render(
      <Button disabled onClick={handleClick}>АВЕ НЛМК</Button>
    );

    const button = screen.getByRole('button', { name: 'АВЕ НЛМК' });
    
    await userEvent.click(button);
    expect(handleClick).not.toHaveBeenCalled();
  })
});
