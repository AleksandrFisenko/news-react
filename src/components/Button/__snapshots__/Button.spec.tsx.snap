// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should be render buttin 1`] = `
<button
  class="_button_0829c1 undefined"
  type="button"
>
  АВЕ НЛМК
</button>
`;

exports[`Button > should be render buttin 2`] = `
<button
  class="_button_0829c1 undefined"
  type="button"
>
  АВЕ НЛМК
</button>
`;
